executor/aggfuncs/func_group_concat.go

```diff
@@ -103,8 +103,11 @@ func (e *groupConcat) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup [
 			p.buffer.WriteString(e.sep)
 		}
 	}
-	p.buffer.Truncate(p.buffer.Len() - len(e.sep))
-	return e.truncatePartialResultIfNeed(sctx, p.buffer)
+	if p.buffer != nil {
```

Comment:
When will the p.buffer be nil?

Reply:
As the test case shows, if all the value of the arg is null, 
p.buffer will be nil.